### PR TITLE
Add Jira support

### PR DESCRIPTION
**Pull request by ricky-rav**
Refactored code and added support for jira queries (bugs and escalations)

Signed-off-by: Riccardo Ravaioli <rravaiol@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,114 +1,90 @@
...
+It fetches open bugs assigned to our team members from our bugzilla and our jira servers, analyzes them and prints out a per-developer summary in the order from the currently least-loaded to currently the most-loaded team member. 
+
+In particular, it queries:
+- bugzilla for existing bugs (the creation new bugs will soon be disabled on our bugzilla server), 
+- jira (OCPBUGS project) for bugs and
+- jira (RHOCPPRIO project) for escalations.
+
+## Download source code
+```
+$ git clone https://github.com/dougsland/bz-query && pushd bz-query
```
Comment (flavio-fernandes) [inline README.md:20]: @ricky-rav I think we need to refine the install a little more.
I tried followoing the steps described here and end up with this error:
```
env ❯ ./network_bugs_overview -h
Traceback (most recent call last):
  File "/home/vagrant/dev/douglas_bugzilla/tmp/bz-query/./network_bugs_overview", line 8, in <module>
    from dateutil import parser
ModuleNotFoundError: No module named 'dateutil'
```

How about we create a file called requirements.txt with the dependencies we need:

```
jira
python-dateutil
```

and the as part of the install add `pip install .`

**Review by flavio-fernandes — CHANGES_REQUESTED** [review]
A few more tweaks to the readme install, please.

File: README.md [hunk 2]
```diff
@@ -1,114 +1,95 @@
...
 
-## First Steps
-First, generate the `api key` to communicate with bugzilla using `python-bugzilla` module.  
-Go to https://bugzilla.redhat.com and click: **Username -> Preferences -> API Keys**.  
+It fetches open bugs assigned to our team members from our bugzilla and our jira servers, analyzes them and prints out a per-developer summary in the order from the currently least-loaded to currently the most-loaded team member. 
 
-**ATENTION:**  
-As soon you click: "**Generate the API-key**" a long string will be generated with chars and numbers, COPY as it's only displayed **ONCE**.
+In particular, it queries:
+- bugzilla for existing bugs (the creation new bugs will soon be disabled on our bugzilla server),
```
Comment (flavio-fernandes) [inline README.md]: nit: white space in the end of this line

**Review by flavio-fernandes — CHANGES_REQUESTED** [review]
hm.... I'm getting this error below. Maybe I'm doing something wrong?

.env ❯ git rev-parse HEAD
eb958cf203c11c4c9209bf9592f0b73d3249cc4a

Also, can you please format the python code with black:
```
pip install black
black network_bugs_overview
```
?

```
.env ❯ ./network_bugs_overview  --jira-bugs
Error running JIRA query project=OCPBUGS and component in ("networking/openshift-sdn", "networking/ovn-kubernetes", "networking") and status in ("NEW", "ASSIGNED", "POST", "ON_DEV", "In Progress"), error: Unterminated string starting at: line 1 column 145102 (char 145101)
- Found 0 bugs with the query: project=OCPBUGS and component in ("networking/openshift-sdn", "networking/ovn-kubernetes", "networking") and status in ("NEW", "ASSIGNED", "POST", "ON_DEV", "In Progress")
  Query processing time: 0.8s
  ```

**Comment by dougsland** [comment]
@ricky-rav @flavio-fernandes awesome patch. +1 for merge it for now. However, as it's getting `pro-level` :)  Should we move to an `official team repo` somewhere?

**Comment by ricky-rav** [comment]
> @ricky-rav @flavio-fernandes awesome patch. +1 for merge it for now. However, as it's getting `pro-level` :) Should we move to an `official team repo` somewhere?

no rush for now, I think :)

**Comment by ricky-rav** [comment]
@flavio-fernandes I ran black on the source file and applied your requested changes.

Also, after clarifying this with the team yesterday, I removed from the default behaviour the queries to the RHOCPPRIO project in Jira. We'll see, as we get more familiar with jira bugs, what escalated bugs in OCPBUGS project will look like, how to include them, and whether we will use RHOCPPRIO issues at all.

**Comment by dougsland** [comment]
looks good :) @flavio-fernandes ?

**Pull request merged**